src/Shouldly/ShouldlyExtensionMethods/ShouldBe/GenericShouldBeTestExtensions.cs

```diff
@@ -12,20 +12,38 @@ public static partial class ShouldBeTestExtensions
         [ContractAnnotation("actual:null,expected:notnull => halt;actual:notnull,expected:null => halt")]
         public static void  ShouldBe<T>(
             [AllowNull, NotNullIfNotNull("expected")] this T actual,
-            [AllowNull, NotNullIfNotNull("actual")] T expected
-            , string? customMessage = null)
+            [AllowNull, NotNullIfNotNull("actual")] T expected,
+            string? customMessage = null)
         {
             if (ShouldlyConfiguration.CompareAsObjectTypes.Contains(typeof(T).FullName!) || typeof(T) == typeof(string))
                 actual!.AssertAwesomely(v => Is.Equal(v, expected, new ObjectEqualityComparer<T>()), actual, expected, customMessage);
             else
                 actual!.AssertAwesomely(v => Is.Equal(v, expected), actual, expected, customMessage);
         }
+        public static void ShouldBe<T>(this T actual, T expected, IEqualityComparer<T> comparer)
+        {
+            ShouldBe(actual, expected, comparer, null);
+        }
```

Comment:
All overloads added in this PR are missing the nullability annotations that are present on the methods they overload.

Reply:
https://github.com/shouldly/shouldly/pull/700